### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by welpaolo**

### CODE REVIEW

File: docs/tutorial/t-manage-passwords.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+This is part of the [Charmed Kafka Tutorial](https://discourse.charmhub.io/t/charmed-kafka-tutorial/). Please refer to this page for more information and the overview of the content.
+
+## Passwords
+
+When we accessed Kafka earlier in this tutorial, we needed to include a password in the connection parameters. 
+Passwords help to secure our cluster and are essential for security. Over time it is a good practice to change the password frequently. Here we will go through setting and changing the password for the admin user.
+
+### Retrieve the admin password
+As previously mentioned, the admin password can be retrieved by running the `get-password` action on the Charmed Kafka application:
```
Comment (zmraul) [inline docs/tutorial/t-manage-passwords.md:11]:
```suggestion
### Retrieve the admin password
As previously mentioned, the admin password can be retrieved by running the `get-admin-credentials` action on the Charmed Kafka application:
```

File: docs/tutorial/t-deploy-kafka.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+This is part of the [Charmed Kafka Tutorial](/t/). Please refer to this page for more information and the overview of the content. 
+
+## Deploy
+
+To deploy Charmed Kafka, all you need to do is run the following command, which will fetch the needed charms([Kafka](https://charmhub.io/kafka?channel=3/edge) and [Zookeeper](https://charmhub.io/zookeeper?channel=edge)) from [Charmhub](https://charmhub.io/) and deploy it to your model:
+
+```shell
+$ juju deploy zookeeper --channel latest/edge -n 5
+$ juju deploy kafka --channel 3/edge -n 3
```
[the remark this answers is not in the document]
Comment (deusebio) [reply docs/tutorial/t-deploy-kafka.md:11]: Actually, we have just released stable now. So we can probably skip the channel information

File: docs/tutorial/t-deploy-kafka.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+The command updates the status of the cluster every second and as the application starts you can watch the status and messages of Charmed Kafka and Zookeeper change. 
+
+Wait until the application is ready - when it is ready, `juju status --watch 1s` will show:
+```shell
+Model     Controller  Cloud/Region         Version  SLA          Timestamp
+tutorial  overlord    localhost/localhost  2.9.38   unsupported  08:20:12Z
+
+App        Version  Status  Scale  Charm      Channel      Rev  Exposed  Message
+kafka               active      3  kafka      3/edge       112  no       
+zookeeper           active      5  zookeeper  latest/edge   98  no
```
Comment (deusebio) [inline docs/tutorial/t-deploy-kafka.md:35]: We could probably update the channel here and the revision

File: docs/tutorial/t-performace-tuning.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
+# Performance Tuning
```
Comment (deusebio) [inline docs/tutorial/t-performace-tuning.md:1]: This does not seem to be a tutorial, but I would rather put this into Reference

File: docs/tutorial/t-overview.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+While this tutorial intends to guide and teach you as you deploy Charmed Kafka, it will be most beneficial if you already have a familiarity with: 
+- Basic terminal commands.
+- Kafka concepts such as replication and users.
+
+## Step-by-step guide
+
+Here’s an overview of the steps required with links to our separate tutorials that deal with each individual step:
+* [Set up the environment](/t/)
+* [Deploy Kafka](/t/)
+* [Managing your units](/t/)
```
Comment (deusebio) [inline docs/tutorial/t-overview.md:26]: Which one is the managing units file?

File: docs/tutorial/t-relate-kafka.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+This is part of the [Charmed Kafka Tutorial](/t/charmed-kafka-tutorial/). Please refer to this page for more information and the overview of the content. 
+
+## Relations
+Relations, or what Juju documentation [describes as Integration](https://juju.is/docs/sdk/integration), are the easiest way to create a user for Kafka in Charmed Kafka. Relations automatically create a username, password, and database for the desired user/application. As mentioned earlier in the [Access Kafka section](#access-kafka) it is a better practice to connect to Kafka via a specific user rather than the admin user.
+
+### Data Integrator Charm
+Before relating to a charmed application, we must first deploy our charmed application. In this tutorial we will relate to the [Data Integrator Charm](https://charmhub.io/data-integrator). This is a bare-bones charm that allows for central management of database users, providing support for different kinds of data platforms (e.g. MongoDB, MySQL, PostgreSQL, Kafka, OpenSearch, etc) with a consistent, opinionated and robust user experience. In order to deploy the Data Integrator Charm we can use the command `juju deploy` we have learned above:
```
Comment (deusebio) [inline docs/tutorial/t-relate-kafka.md:9]:
```suggestion
To start using the Kafka cluster, we will now relate our application to the [Data Integrator Charm](https://charmhub.io/data-integrator). This is a bare-bones charm that allows for central management of database users, providing support for different kinds of data platforms (e.g. MongoDB, MySQL, PostgreSQL, Kafka, OpenSearch, etc) with a consistent, opinionated and robust user experience. In order to deploy the Data Integrator Charm we can use the command `juju deploy` we have learned above:
```

Comment (deusebio) [reply docs/tutorial/t-relate-kafka.md:9]: We are telling a story and the tutorial is supposed to be done from top to bottom. We don't have to tell the user that we must first kafka.

File: docs/tutorial/t-relate-kafka.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+```shell
+juju deploy data-integrator --channel stable --config topic-name=test-topic --config extra-user-roles=producer,consumer
+```
+The expected output:
+```shell
+Located charm "data-integrator" in charm-hub, revision 11
+Deploying "data-integrator" from charm-hub charm "data-integrator", revision 11 in channel stable on jammy
+```
+
+### Relate to MongoDB
```
Comment (deusebio) [inline docs/tutorial/t-relate-kafka.md:20]:
```suggestion
### Relate to Kafka
```

File: docs/tutorial/t-setup-environment.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
+# Environment Setup
+
+This is part of the [Charmed Kafka Tutorial](/t/). Please refer to this page for more information and the overview of the content. 
+
+## Requirements
+
+For production environments, it is recommended to deploy at least 5 nodes for Zookeeper and 3 for Kafka. While the following requirements are meant to be for production, the charm can be deployed in much smaller environments.
```
Comment (deusebio) [inline docs/tutorial/t-setup-environment.md:7]: If we are in the tutorial, it is not useful to list "prod" requirements. 

I would revised these requirements to what needed for the tutorial, and then place the Prod requirements, into a separated section in "Reference/requirements.md"

**Review by deusebio — APPROVED** [review]
Looks good! Just few niptick to be addressed. Once the content is stable, I can migrate this to Discourse BEFORE merging the PR, making sure that the PR is inline with Discourse.

File: docs/tutorial/t-deploy-kafka.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (zmraul) [reply docs/tutorial/t-deploy-kafka.md:11]: Updated the suggestion

File: docs/tutorial/t-deploy-kafka.md (hunk 3 above)
Comment (welpaolo) [reply docs/tutorial/t-deploy-kafka.md:35]: Done

**Pull request merged**